python/triton/runtime/interpreter.py

```diff
@@ -1051,12 +1061,22 @@ def __init__(self, fn, arg_names, grid):
         self.constexprs = [name for name in arg_names if __annotations__.get(name) == "constexpr"]
 
     def _init_args_hst(self, args_dev, kwargs):
-        args_hst = []
-        for arg in args_dev:
-            if hasattr(arg, "data_ptr"):
-                args_hst.append(arg.cpu())
-            else:
-                args_hst.append(arg)
+        def _to_cpu(arg):
+            if not hasattr(arg, "data_ptr"):
+                return arg
+            unwrapped_arg = _unwrap_tensor(arg)
+            cpu_arg = unwrapped_arg.new_empty(unwrapped_arg.untyped_storage().size() // unwrapped_arg.element_size(), device='cpu')
```

Comment:
NIT: Since you don't actually use the tensor's storage, this can be an empty allocation:
```suggestion
            cpu_arg = unwrapped_arg.new_empty(0, device='cpu')
```

Reply:
Oh that's great! it makes the code there a lot more readable too